File: Wifi_scheme_management_application/angularapp/src/app/services/wifi-scheme-request.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { WifiSchemeRequest } from '../models/wifi-scheme-request.model';
import { Observable } from 'rxjs';
import { APP_URL } from '../app.constants';

@Injectable({ providedIn: 'root' })
export class WifiSchemeRequestService {

  // public apiUrl=`http://localhost:8080/api/wifiSchemeRequest`;


  constructor(private readonly http: HttpClient) {}

  getAllWiFiSchemeRequests(): Observable<WifiSchemeRequest[]> {
    return this.http.get<WifiSchemeRequest[]>(APP_URL+"/wifiSchemeRequest");
  }

  getWiFiSchemeRequestsByUserId(userId: number): Observable<WifiSchemeRequest[]> {
    return this.http.get<WifiSchemeRequest[]>(`${APP_URL+"/wifiSchemeRequest"}/user/${userId}`);
  }

  addWiFiSchemeRequest(request: WifiSchemeRequest): Observable<WifiSchemeRequest> {
    console.log(request);
    
    return this.http.post<WifiSchemeRequest>(`${APP_URL+"/wifiSchemeRequest"}`, request);
  }

  updateWiFiSchemeRequest(id: number, request: WifiSchemeRequest): Observable<WifiSchemeRequest> {
    return this.http.put<WifiSchemeRequest>(`${APP_URL+"/wifiSchemeRequest"}/${id}`, request);
  }

  deleteWiFiSchemeRequest(id: number): Observable<any> {
    return this.http.delete<void>(`${APP_URL+"/wifiSchemeRequest"}/${id}`);
  }
}

File: Wifi_scheme_management_application/springapp/src/main/java/com/examly/springapp/service/WiFiSchemeRequestService.java
package com.examly.springapp.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.examly.springapp.model.WiFiSchemeRequest;


@Service
public interface WiFiSchemeRequestService {

    WiFiSchemeRequest addWiFiSchemeRequest(WiFiSchemeRequest request);
    List<WiFiSchemeRequest> getWiFiSchemeRequestsByUserId(Long userId);
    List<WiFiSchemeRequest> getAllWiFiSchemeRequests();
    WiFiSchemeRequest updateWiFiSchemeRequest(Long id, WiFiSchemeRequest request);
    void deleteWiFiSchemeRequest(Long id);
    WiFiSchemeRequest getWiFiSchemeRequestById(Long id);

}